cpp/include/cudf/types.h

```diff
@@ -85,14 +85,32 @@ typedef struct {
 	// here we can also hold info for decimal datatype or any other datatype that requires additional information
 } gdf_dtype_extra_info;
 
+/**---------------------------------------------------------------------------*
+ * @union gdf_data
+ * @brief Union used for scalar type.
+ * It stores a unique value for scalar type.
+ * It has a direct relationship with the gdf_dtype.
+ *---------------------------------------------------------------------------**/
+typedef union {
+  int8_t   si08;  /**< GDF_INT8      */
+  int16_t  si16;  /**< GDF_INT16     */
+  int32_t  si32;  /**< GDF_INT32     */
+  int64_t  si64;  /**< GDF_INT64     */
+  float    fp32;  /**< GDF_FLOAT32   */
+  double   fp64;  /**< GDF_FLOAT64   */
+  int32_t  dt32;  /**< GDF_DATE32    */
```

Comment:
These will need to use the wrapper types.

Reply:
Then it won't be a C only header anymore. Is that OK?

Reply:
https://github.com/rapidsai/cudf/pull/577 adds a `types.hpp` file if you'd like to put it there instead. Otherwise, if you can add the wrapper types here and everything still builds, then there's no problem with making it a non-C file.

Reply:
It didn't build as I suspected. CFFI complains that it cannot parse this header. If I put this in `types.hpp` then `gdf_scalar` would not be defined in python land. If I just put the traits in `types.hpp` and the `gdf_scalar` definition in `types.h` then I'd need to include `types.hpp` in `types.h` which will again cause the parsing error. This issue should go away when we drop cffi.

Reply:
If this is not a deal breaker, I can add a TODO here.

Reply:
`TODO` is sufficient. Don't let it block the PR.